pulsar/client/staging/up.py

```diff
@@ -207,7 +207,8 @@ def __initialize_referenced_arbitrary_files(self):
 
     def __upload_tool_files(self):
         for tool_file in self.tool_files:
-            self.transfer_tracker.handle_transfer_path(tool_file, path_type.TOOL)
+            name = self.job_inputs.relative_file_name(self.tool_dir, tool_file)
```

Comment:
This looks a lot better to my eye - I still don't understand why this needed for tools but not for say UNSTRUCTURED files (``__upload_arbitrary_files``). That is really odd to me - but at least the hack is more limited now.